src/adapter/src/catalog/apply.rs

```diff
@@ -299,28 +301,24 @@ impl CatalogState {
     ) -> Vec<BuiltinTableUpdate<&'static BuiltinTable>> {
         let mut builtin_table_updates = Vec::with_capacity(updates.len());
         for StateUpdate { kind, diff } in updates {
-            if diff == -1 {
-                builtin_table_updates
-                    .extend(self.generate_builtin_table_update(kind.clone(), diff));
-                self.apply_update(kind, diff);
-            } else if diff == 1 {
-                self.apply_update(kind.clone(), diff);
-                builtin_table_updates
-                    .extend(self.generate_builtin_table_update(kind.clone(), diff));
-            } else {
-                unreachable!("invalid update in catalog updates: ({kind:?}, {diff:?})")
+            match diff {
+                StateDiff::Retraction => {
+                    builtin_table_updates
+                        .extend(self.generate_builtin_table_update(kind.clone(), diff));
+                    self.apply_update(kind, diff);
+                }
+                StateDiff::Addition => {
+                    self.apply_update(kind.clone(), diff);
+                    builtin_table_updates
+                        .extend(self.generate_builtin_table_update(kind.clone(), diff));
+                }
```

Comment:
Seemingly the ordering of `apply_update` and `builtin_table_updates.extend(...)` matters, do you mind leaving a note here as to why?

Reply:
Done